src/repl/ast/parser/mod.rs

```diff
@@ -0,0 +1,2384 @@
+                        e.span = span;
+                        Some(e)
+                    }
+                    Token::Whitespace => {
+                        let name = ClarityName::try_from("<").unwrap();
+                        let mut e = PreSymbolicExpression::atom(name);
+                        e.span = token.span;
+                        Some(e)
+                    }
+                    _ => {
+                        // Pretend there is an identifier here and try to continue.
+                        let mut span = token.span.clone();
+                        let unexpected = self.next_token().unwrap();
+                        self.add_diagnostic(PlacedError {
+                            e: ParserError::UnexpectedToken(unexpected.token.clone()),
+                            span: unexpected.span.clone(),
+                        });
+                        let cname = ClarityName::try_from("placeholder".to_string()).unwrap();
+                        match self.tokens[self.next_token].token {
+                            Token::Greater => {
+                                let close = self.next_token().unwrap();
+                                span.end_line = close.span.end_line;
+                                span.end_column = close.span.end_column;
+                            }
+                            _ => {
+                                self.add_diagnostic(PlacedError {
+                                    e: ParserError::ExpectedClosing(Token::Greater),
+                                    span: self.tokens[self.next_token].span.clone(),
+                                });
+                                self.add_diagnostic(PlacedError {
+                                    e: ParserError::NoteToMatchThis(token.token),
+                                    span: token.span.clone(),
+                                });
+                                span.end_line = unexpected.span.end_line;
+                                span.end_column = unexpected.span.end_column;
+                            }
+                        }
+                        let mut e = PreSymbolicExpression::trait_reference(cname);
+                        e.span = span;
+                        Some(e)
+                    }
+                }
+            }
+            Token::Plus
+            | Token::Minus
+            | Token::Multiply
+            | Token::Divide
+            | Token::LessEqual
+            | Token::Greater
+            | Token::GreaterEqual => {
+                let name = ClarityName::try_from(format!("{}", token.token)).unwrap();
+                let mut e = PreSymbolicExpression::atom(name);
+                e.span = token.span;
+                Some(e)
+            }
+            // TODO: For now, comments are being treated as whitespace. In the future,
+            //       they should be attached to the following expression
+            // Token::Comment(comment) => {
```

Comment:
We could handle the comments and collect these mini-ast in a side structure, if you wanted.

Reply:
Yes, I like that. Are you okay with that going into a separate PR after the initial set of changes in merged in? I'll open an issue for that.

Reply:
Opened #88.